sphinx/util/logging.py

```diff
@@ -324,22 +325,8 @@ def pending_logging() -> Iterator[MemoryHandler]:
 
 @contextmanager
 def skip_warningiserror(skip: bool = True) -> Iterator[None]:
-    """Context manager to skip WarningIsErrorFilter temporarily."""
-    logger = logging.getLogger(NAMESPACE)
-
-    if skip is False:
-        yield
-    else:
-        try:
-            disabler = DisableWarningIsErrorFilter()
-            for handler in logger.handlers:
-                # use internal method; filters.insert() directly to install disabler
-                # before WarningIsErrorFilter
-                handler.filters.insert(0, disabler)
-            yield
-        finally:
-            for handler in logger.handlers:
-                handler.removeFilter(disabler)
+    # Deprecate in Sphinx 10
```

Comment:
Do you want it to be deprecated in Sphinx 10 and removed in 12, or removed in 10 and deprecated now? I'd suggest deprecating the call `skip=True` now so that users are notified of this change (and add it to the list of deprecated objects).

Reply:
The former, deprecating immediatley is often useful but can be annoying with things like flags. This gives a longer transitional window.